airflow/models/dataset.py

```diff
@@ -275,6 +275,8 @@ class DatasetEvent(Base):
     source_dag_id = Column(StringID(), nullable=True)
     source_run_id = Column(StringID(), nullable=True)
     source_map_index = Column(Integer, nullable=True, server_default=text("-1"))
+    external_service_id = Column(StringID(), nullable=True)
+    external_source = Column(StringID(), nullable=True)
```

Comment:
How are these different and why can they not be combined into one field?

Reply:
As per earlier comment from me @uranusjr :

* by whom - external_auth_id or external_service_id -> required
* from where (api, client_ip / remote_addr) - external_source -> required

So they cannot be combined as they contain different information and are both required.